langchain4j/src/main/java/dev/langchain4j/service/AiServiceContext.java

```diff
@@ -24,6 +26,9 @@ public class AiServiceContext {
 
     public ToolService toolService = new ToolService();
 
+    public final GuardrailService.Builder guardrailServiceBuilder;
+    private final AtomicReference<GuardrailService> guardrailService = new AtomicReference<>();
```

Comment:
I'm not fully understanding this design: can you please clarify why you need to keep an instance of the builder here and then (lazily?) create the `GuardrailService` out of it instead of just having the service alone in this context?

Reply:
I was trying to keep the execution of the guardrails (i.e. the `GuardrailService`) separate from the construction of the "what" needed to be executed. Since the guardrails can be constructed in many different ways (annotations on class, annotations on methods, progamatically via the `AiServices` builder) I wanted somewhat of a "holder" of the information that could then be combined to build the service which actually executes the guardrails.

I didn't want to put the logic of constructing "what" to execute vs the actual execution logic.